# STAT 207 Lab 7: Hypothesis Testing - [25.5 total points] (0.5) is bonus points - Solutions

Due: Wednesday, March 24 11:59pm CST



## <u>Case Study</u>: U.S Unemployment Rates and Median Education Levels
In this lab we will explore the statistical concepts that we've learned this week while also examining the **unemployment rates** and **median education levels** of the counties in the U.S. 

### Imports

In [64]:
import pandas as pd                    # imports pandas and calls the imported version 'pd'
import matplotlib.pyplot as plt        # imports the package and calls it 'plt'
import seaborn as sns                  # imports the seaborn package with the imported name 'sns'
sns.set()                              # sets the default plotting style
from scipy.stats import norm
import numpy as np

## 1. <u>Preliminaries</u>

### Read the county.csv into a dataframe, call it df, and display the first five rows.

Remember that 'data unavailable' represents missing values in this csv. Make sure you encode these values as NaN when you read the data in.

In [7]:
missing_values = ["data unavailable"]
df = pd.read_csv("county.csv", na_values = missing_values)
df.head()

name    state   pop2000  pop2010   pop2017  pop_change  poverty  \
0  Autauga County  Alabama   43671.0    54571   55504.0        1.48     13.7   
1  Baldwin County  Alabama  140415.0   182265  212628.0        9.19     11.8   
2  Barbour County  Alabama   29038.0    27457   25270.0       -6.22     27.2   
3     Bibb County  Alabama   20826.0    22915   22668.0        0.73     15.2   
4   Blount County  Alabama   51024.0    57322   58013.0        0.68     15.6   

   homeownership  multi_unit  unemployment_rate metro    median_edu  \
0           77.5         7.2               3.86   yes  some_college   
1           76.7        22.6               3.99   yes  some_college   
2           68.0        11.1               5.90    no    hs_diploma   
3           82.9         6.6               4.39   yes    hs_diploma   
4           82.0         3.7               4.02   yes    hs_diploma   

   per_capita_income  median_hh_income smoking_ban  
0           27841.70           55317.0        none  
1           27779.85           52562.0        none  
2           17891.73           33368.0     partial  
3           20572.05           43404.0        none  
4           21367.39           47412.0        none

### Create a series that is comprised of just the unemployment rate variable and drop any missing values from this dataset.

In [8]:
df_unemp_rate=df['unemployment_rate'].dropna()

### Create a series that is comprised of just the median education variable and drop any missing values from this dataset.

In [9]:
df_med_edu=df['median_edu'].dropna()

## 2. <u>Unemployment Rate Population Analysis</u>

We will assume that the series of all county unemployment rates from 1b is our **population**. Usually, we do not have the whole population at our disposal, which is why we conduct inference in the first place.

However, when conducting our hypothesis testing in this assignment we will assume that we don't have this population information. After we have conducted the hypothesis testing, we can use our population information to "check" our answers to learn more about how hypothesis testing works.

### 2a. [0.5 pt] Calculate the mean and standard deviation unemployment rate of ALL U.S. counties.

In [10]:
df_unemp_rate.mean()

4.611443134756292

In [11]:
df_unemp_rate.std()

1.648761085622538

### 2b.  [1 pt] Calculate the z-score of the unemployment rate of 'Champaign County' in the state of 'Illinois'.

In [51]:
df.loc[604]

name                 Champaign County
state                        Illinois
pop2000                      179669.0
pop2010                        201081
pop2017                      209399.0
pop_change                       1.83
poverty                          21.4
homeownership                    55.9
multi_unit                       35.2
unemployment_rate                4.24
metro                             yes
median_edu               some_college
per_capita_income            29619.49
median_hh_income              49586.0
smoking_ban                       NaN
Name: 604, dtype: object

In [46]:
print((4.24 - df_unemp_rate.mean()) / df_unemp_rate.std())

-0.2252862091392959


## 3. <u>Unemployment Rate Sample Analysis</u>

### 3a.  [0.5 pt] Collect a random sample of size 35 from your population of all county unemployment rates (without replacement) using a random state of 3.

In [48]:
sample = df_unemp_rate.sample(35, random_state = 3, replace = False)

### 3b.  [0.5 pt] Calculate the mean and standard deviation of this sample.

In [49]:
sample.mean()

4.252

In [50]:
sample.std()

1.3095751443974222

## 4. <u>Unemployment Rate Sampling Distribution Analysis</u>

Assume you know $\sigma$ in this part.

### 4a.  [0.5 pt] What is the mean and standard deviation of the sampling distribution of sample mean unemployment rates (collected from samples of size 35)?

The mean of the sampling distribution is the same as the population mean

In [52]:
df_unemp_rate.mean()

4.611443134756292

The standard deviation of the sampling distribution is the population standard deviation devided by the square root of n

In [54]:
print(df_unemp_rate.std() / (35 ** 0.5))

0.2786914893089392


### 4b.  [1 pt] Is the sampling distribution of samples means (collected from samples of size 35) approximately normal? Explain why or why not?

Yes, by central limit theorum we know that the sampling distribution of sample means is approximately normal because n > 30.

### 4c.  [1 pt] What is the z-score of our sample mean from 3b?

In [55]:
print((sample.mean() - df_unemp_rate.mean()) / df_unemp_rate.std())

-0.21800801698359698


### 4d.  [1 pt] What is the probability that a randomly selected sample mean z-score is greater than or equal to the z-score that we calculated in 4c?

In [59]:
1 - norm.cdf((sample.mean() - df_unemp_rate.mean()) / df_unemp_rate.std())

0.5862885674956217

### 4e.  [0.5 pt] What is the probability that a randomly selected sample mean z-score is less than or equal to the negative absolute value of the z-score that we calculated in 4c?

In [60]:
norm.cdf((sample.mean() - df_unemp_rate.mean()) / df_unemp_rate.std())

0.4137114325043782

## 5. <u>Hypothesis Testing for the Average Unemployment Rate of ALL US Counties</u>

For now, suppose that we know nothing about our population that we explored in #2.


Suppose that a reliable source suggested that the average unemployment rate of all US counties is 4.611. We are skeptical of this source, and would like to test this claim and see if there is evidence to suggest that this is not the case. We will use our random sample that we collected from the population in #3 to test this claim.

### 5a.  [1 pt] Formulate your null and alternative hypothses for this hypothesis test.

null: mu = 4.611
alternate: mu != 4.611

### 5b.  [1 pt] Check the conditions for conducting hypothesis testing using the sample from #3.



1) The observations are independent because the sample is collected randomly and 35 is less than 10% of all counties in the U.S.
2) n > 30 so we know the sampling distribution is approximately normal


### 5c.  [1 pt] Calculate the p-value for this test.

Assume you do not know $\sigma$.

In [61]:
assumed_pop_mean = 4.611
print('Assumed Pop Mean is: ', assumed_pop_mean)

Assumed Pop Mean is:  4.611


In [65]:
std_error=(sample.std()/(np.sqrt(35)))
print('Standard Error: ', std_error)

Standard Error:  0.22135860103483856


In [69]:
pvalue= 2*(norm.cdf(sample.mean(), loc=assumed_pop_mean, scale=std_error))
print('p-value: ', pvalue)

p-value:  0.10484557772753991


### 5d.  [1 pt] Make a conclusion about your hypotheses using this p-value and a significance level of $\alpha=0.05$.

Because the p-value of 0.105 >= a = 0.05, we fail to reject the null hypothesis. We do not have sufficient evidence to suggest the alternative hypothesis that the average unemployment rate of alll US counties is different that 4.611. 

### 5e.  [1 pt] Now we would like to make a 95% confidence interval for the average unemployment rate of al U.S. counties. First, check the conditions for creating this confidence interval.

We are allowed to make a confidence interval for mu because the following conditions for CLT are met:
1) The observations are independent because the sample is collected randomly and 35 is less than 10% of all counties in the U.S.
2) n > 30 so we know the sampling distribution is approximately normal

### 5f.  [1 pt] Now create this 95% confidence interval.

Assume you do not know $\sigma$.

In [73]:
critical_value = norm.ppf(0.975)
critical_value
lower_bound=sample.mean()-critical_value*(sample.std()/np.sqrt(35))
upper_bound=sample.mean()+critical_value*(sample.std()/np.sqrt(35))
print(lower_bound, ',', upper_bound)

3.8181451143035456 , 4.685854885696454


### 5g.  [1 pt] Use this confidence interval to make a conclusion about your hypothesis tests.

Since the value '4.611' resides in the interval given above, we are 95% confident that the true average unemployment rate of US counties is in between 3.8181451143035456 , 4.685854885696454. Therefore, we fail to reject the null hypothesis. We do not have sufficient evidence to suggest the alternative hypothesis that the average unemployment rate of alll US counties is different that 4.611. 

### 5h.  [0.5 pt] NOW, using what we *actually* know about the population and population mean, did our hypothesis test *actually* make the right conclusions in 5d and 5g?

Yes, our hypothesis test did make the right conclusions in 5d and 5g because the true mean is roughly 4.611, and both of these tests failed to reject the null.

## 6. <u>Median Education Level Population Analysis</u>

We will assume that the series of all county median education levels from 1c is our **population**. Usually, we do not have the whole population at our disposal, which is why we conduct inference in the first place.

However, when conducting our hypothesis testing in this assignment we will assume that we don't have this population information. After we have conducted the hypothesis testing, we can use our population information to "check" our answers to learn more about how hypothesis testing works.



### <u>Tutorial</u>: Conditional Filtering of a Series

In [4]:
import pandas as pd
import numpy as np
small_series=pd.Series(['stat107', 'stat207', 'stat307', 'stat207'])
small_series

0    stat107
1    stat207
2    stat307
3    stat207
dtype: object

We can count the number of entries in a series that meet a certain condition by setting up a condition upon the values in a series and taking the sum of the Boolean values.

In [5]:
small_series=='stat207'

0    False
1     True
2    False
3     True
dtype: bool

In [6]:
np.sum(small_series=='stat207')

2

### 6a.  [0.5 pt] Calculate the proportion of all U.S. counties that have a median education level of 'hs_diploma'.

In [81]:
pop_prop = np.sum(df_med_edu=='hs_diploma')/df.shape[0]
pop_prop

0.4446212603437301

## 7. <u>Median Population Sample Analysis</u>

### 7a.  [0.5 pt] Collect a random sample of 40 county median education levels from the population of ALL U.S. county median education levels, (without replacement), using a random state of 100.

In [78]:
sample2 = df_med_edu.sample(40, random_state = 100, replace = False)

### 7b.  [0.5 pt] Calculate the proportion of counties in this sample that have a median education level of high school diploma.

In [80]:
np.sum(sample2=='hs_diploma')/40

0.5

## 8. <u>Sampling Distribution of Sample Proportions Analysis</u>



### 8a. [1 pt] What is the mean and standard deviation of the sampling distribution of sample propotion of counties with a median education level of high school diploma (collected from samples of size 40)?

Assume you actually know the population proprtion $p$ in part 8.

The mean is equal to p. 

In [82]:
print(pop_prop)

0.4446212603437301


The std is equal to the given formula: sqrt(p(1-p)/n)

In [84]:
print(np.sqrt((pop_prop*(1-pop_prop))/40))

0.07857054078884831


### 8b.  [1 pt] What is the z-score of our sample proportion from 7b?

In [85]:
print((0.5 - pop_prop) / (np.sqrt((pop_prop*(1-pop_prop))/40)))

0.7048282867887541


## 9. <u>Hypothesis Testing for the Proportion of ALL U.S. Counties that have a Median Education Level of High School Diploma</u>

For now, suppose that we know nothing about our population that we explored in #6.

Previous studies have suggested that in the past, 45% of U.S. counties have a median education level of a high school diploma. We are skeptical that that is the still the case. We would like to test (using our sample from #7), whether there is evidence to suggest that the the proportion of counties with a median education level of high school diploma is diffrent from 45%.

### 9.a.  [1 pt] Formulate your null and alternative hypothses for this hypothesis test.

null: p = .45
alternate: p != .45

### 9b.  [1 pt] Check the conditions for conducting hypothesis testing using the sample from #7.



1) We know the observations are independent because the sample is randomly collected and 40 is less than 10% of the population of all counties in the US.
2)
np = 40(.45) = 18 >= 10
n(1-p) = 40(1-.45) = 22 >= 10


### 9c.  [1 pt] Calculate the p-value for this test.

Assume you do not know anything about the population proportion (including the value of $p$).

In [86]:
assumed_pop_prop=0.45
print('Asssumed Pop Prop: ', assumed_pop_prop)

Asssumed Pop Prop:  0.45


In [87]:
std_error=np.sqrt(assumed_pop_prop*(1-assumed_pop_prop)/40)
print('Std Error: ', std_error)

Std Error:  0.07866066361276136


In [89]:
pvalue=2*(1 - norm.cdf(0.5, loc=assumed_pop_prop, scale=std_error))
print('Pvalue: ', pvalue)

Pvalue:  0.5250099664186034


### 9d.  [1 pt] Make a conclusion about your hypotheses using this p-value and a significance level of $\alpha=0.10$.

Since p-value 0.525 > a = 0.10, we fail to reject the null hypothesis. We do not have sufficient evidence to suggest the alternative hypothesis that the proportion of counties with a median education of 'hs_diploma' of all US counties is different that 0.45. 

### 9e.  [1 pt] Now we would like to make a 90% confidence interval for the proportion of ALL U.S. counties that have a median education level of high school diploma.. First, check the conditions for creating this confidence interval.

1) We know the observations are independent because the sample is randomly collected and 40 is less than 10% of the population of all counties in the US.
2)
np = 40(.45) = 18 >= 10
n(1-p) = 40(1-.45) = 22 >= 10


### 9f.  [2 pt] Now create this 90% confidence interval.

Assume you do not know anything about the population (including $p$).

In [96]:
critical_value=norm.ppf(1-0.1/2)
lower_bound = .5 - critical_value * np.sqrt(assumed_pop_prop*(1-assumed_pop_prop)/40)
upper_bound = .5 + critical_value * np.sqrt(assumed_pop_prop*(1-assumed_pop_prop)/40)
print(lower_bound, ',', upper_bound)

0.3706147221581398 , 0.6293852778418603


### 9g.  [1 pt] Use this confidence interval to make a conclusion about your hypothesis tests.

Since the value 0.45 resides in the interval above, we fail to reject the null hypothesis. We do not have sufficient evidence to suggest the alternative hypothesis that the proportion of counties with a median education of 'hs_diploma' of all US counties is different that 0.45. 